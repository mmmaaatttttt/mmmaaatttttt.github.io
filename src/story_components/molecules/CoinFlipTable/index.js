import React, { Component } from "react";
import { ColoredSpan, LabeledSlider, StyledTable } from "story_components";
import withCaption from "hocs/withCaption";
import COLORS from "utils/styles";

class CoinFlipTable extends Component {
  state = {
    headsProb: 0.5
  };

  handleProbChange = val => {
    this.setState({ headsProb: val });
  };

  format = (probability, dec = 0) => `${(probability * 100).toFixed(dec)}%`;

  render() {
    const { headsProb } = this.state;
    const tailsProb = 1 - headsProb;
    const pairProb = headsProb * tailsProb;
    const formatted = {
      heads: this.format(headsProb),
      tails: this.format(tailsProb),
      pair: this.format(pairProb, 2)
    };
    return (
      <div>
        <LabeledSlider
          min={0.01}
          max={0.99}
          step={0.01}
          value={headsProb}
          handleValueChange={this.handleProbChange}
          title={`Probability of flipping heads: ${formatted.heads}`}
          color={COLORS.GREEN}
          minIcon="times-circle"
          maxIcon="check-circle"
        />
        <StyledTable margin="0.72rem 0 0 0">
          <thead>
            <tr>
              <th>Prob. of H</th>
              <th>Prob. of T</th>
              <th>Prob. of HT</th>
              <th>Prob. of TH</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>{formatted.heads}</td>
              <td>{formatted.tails}</td>
              <td>
                {formatted.heads} &times; {formatted.tails} ={" "}
                <ColoredSpan color={COLORS.GREEN} bold>{formatted.pair}</ColoredSpan>
              </td>
              <td>
                {formatted.tails} &times; {formatted.heads} ={" "}
                <ColoredSpan color={COLORS.GREEN} bold>{formatted.pair}</ColoredSpan>
              </td>
            </tr>
          </tbody>
        </StyledTable>
      </div>
    );
  }
}

CoinFlipTable.propTypes = {};

CoinFlipTable.defaultProps = {};

export default withCaption(CoinFlipTable);

export { CoinFlipTable };
