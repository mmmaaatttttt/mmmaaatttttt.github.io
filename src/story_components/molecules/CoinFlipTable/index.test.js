import React from "react";
import { render } from "@testing-library/react";
import CoinFlipTable from "./";

it("renders without crashing", () => {
  render(<CoinFlipTable />);
});

it("matches snapshot for a fair coin", () => {
  const { asFragment } = render(<CoinFlipTable />);
  expect(asFragment()).toMatchSnapshot();
});
