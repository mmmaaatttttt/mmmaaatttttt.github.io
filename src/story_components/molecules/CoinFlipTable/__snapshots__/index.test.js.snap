// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests matches snapshot for fair coin 1`] = `
<div>
  <LabeledSlider
    color="#52A081"
    fadeIcons={false}
    handleValueChange={[Function]}
    max={0.99}
    maxIcon="check-circle"
    min={0.01}
    minIcon="times-circle"
    sliderHeight={6}
    sliderPadding={10}
    step={0.01}
    tickCount={2}
    title="Probability of flipping heads: 50%"
    value={0.5}
  />
  <styled.table
    margin="0.72rem 0 0 0"
  >
    <thead>
      <tr>
        <th>
          Prob. of H
        </th>
        <th>
          Prob. of T
        </th>
        <th>
          Prob. of HT
        </th>
        <th>
          Prob. of TH
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          50%
        </td>
        <td>
          50%
        </td>
        <td>
          50%
           × 
          50%
           =
           
          <styled.span
            bold={true}
            color="#52A081"
          >
            25.00%
          </styled.span>
        </td>
        <td>
          50%
           × 
          50%
           =
           
          <styled.span
            bold={true}
            color="#52A081"
          >
            25.00%
          </styled.span>
        </td>
      </tr>
    </tbody>
  </styled.table>
</div>
`;

exports[`smoke and snapshot tests matches snapshot for unfair coin 1`] = `
<div>
  <LabeledSlider
    color="#52A081"
    fadeIcons={false}
    handleValueChange={[Function]}
    max={0.99}
    maxIcon="check-circle"
    min={0.01}
    minIcon="times-circle"
    sliderHeight={6}
    sliderPadding={10}
    step={0.01}
    tickCount={2}
    title="Probability of flipping heads: 50%"
    value={0.5}
  />
  <styled.table
    margin="0.72rem 0 0 0"
  >
    <thead>
      <tr>
        <th>
          Prob. of H
        </th>
        <th>
          Prob. of T
        </th>
        <th>
          Prob. of HT
        </th>
        <th>
          Prob. of TH
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          50%
        </td>
        <td>
          50%
        </td>
        <td>
          50%
           × 
          50%
           =
           
          <styled.span
            bold={true}
            color="#52A081"
          >
            25.00%
          </styled.span>
        </td>
        <td>
          50%
           × 
          50%
           =
           
          <styled.span
            bold={true}
            color="#52A081"
          >
            25.00%
          </styled.span>
        </td>
      </tr>
    </tbody>
  </styled.table>
</div>
`;
